std/assembly/typedarray.ts

```diff
@@ -1593,12 +1549,12 @@ function FILTER<TArray extends ArrayBufferView, T>(
     }
   }
   // shrink output buffer
-  var length = j << alignof<T>();
-  var data = __realloc(buffer, length);
-  out.data = changetype<ArrayBuffer>(data); // retain
-  out.dataStart = data;
-  out.dataLength = length;
-  return out;
+  var byteLength = j << alignof<T>();
+  var data = __realloc(buffer, byteLength);
+  store<ArrayBuffer>(out, changetype<ArrayBuffer>(data), offsetof<TArray>("buffer"));
```

Comment:
appears to miss a `__retain`

Reply:
essentially, whenever a managed reference is assigned to a field, the compiler would first `__retain` the new value and then `__release` the old value, with the second step only relevant if the previous value isn't `null` (like when allocating a new array).

Reply:
Yeah. thanks!